### ISSUE TITLE
PEAR.Commenting.FunctionComment does not support intersection types in comments

### ISSUE THREAD

**Issue by Stadly** [squizlabs/PHP_CodeSniffer#2530]
`Squiz.Commenting.FunctionCommentSniff` does not seem to support intersection types.

```php
/**
 * @param (TranslatorInterface&LocaleAwareInterface)|null $translator Translator for translating messages.
 */
public function setTranslator(?TranslatorInterface $translator): void
{
    $this->translator = $translator;
}
```

Incorrectly generates the following errors:

```
  4 | ERROR | Doc comment for parameter &LocaleAwareInterface)|null does not match actual variable name
    |       | $translator (Squiz.Commenting.FunctionComment.ParamNameNoMatch)
  4 | ERROR | Parameter comment must start with a capital letter
    |       | (Squiz.Commenting.FunctionComment.ParamCommentNotCapital)
```

**Comment by jrfnl** [comment]
This is addressed in PR #2456 and should be fixed once that PR has been merged.

**Comment by gsherwood** [comment]
Thanks for the report. I've added a fix for this specific sniff.
